CrimeMonitor/CrimeMonitor.cs

```diff
@@ -0,0 +1,849 @@
+                    homeSystem = FindHomeSystem(record.faction, factionSystems)
+                        ?? presences.OrderByDescending(p => p.influence).First().systemName;
+                }
+            }
+            // If 'home system' is desiginated, check if system is part of faction presence
+            else if (factionSystems.Contains(homeSystem))
+            {
+                if (FindHomeSystem(record.faction, factionSystems) == null && !homeSystems.ContainsKey(record.faction))
+                {
+                    // Save home system if not part of faction name and not previously saved
+                    homeSystems.Add(record.faction, homeSystem);
+                }
+            }
+            // System not found, exit.
+            else { return; }
+
+            StarSystem factionStarSystem = StarSystemSqLiteRepository.Instance.GetOrFetchStarSystem(homeSystem);
+            record.system = homeSystem;
+
+            if (factionStarSystem != null)
+            {
+                // Filter stations within the faction system which meet the game version and landing pad size requirements
+                 string shipSize = EDDI.Instance?.CurrentShip?.size ?? "Large";
+                List<Station> factionStations = EDDI.Instance.inHorizons ? factionStarSystem.stations : factionStarSystem.orbitalstations
+                    .Where(s => s.stationservices.Count > 0 && s.LandingPadCheck(shipSize)).ToList();
+
+                // Prioritize controlled stations
+                List<Station> controlledStations = factionStations.Where(s => s.Faction.name == record.faction).ToList();
+                if (controlledStations.Count > 0 && controlledStations.Min(s => s.distancefromstar) < 10000)
+                {
+                    factionStations = controlledStations;
+                }
+
+                // Build list to find the faction station nearest to the main star
+                SortedList<decimal, string> nearestList = new SortedList<decimal, string>();
+                foreach (Station station in factionStations)
+                {
+                    if (!nearestList.ContainsKey(station.distancefromstar ?? 0))
+                    {
+                        nearestList.Add(station.distancefromstar ?? 0, station.name);
+                    }
+                }
+
+                // Faction station nearest to the main star
+                string nearestStation = nearestList.Values.FirstOrDefault();
+                record.station = nearestStation;
+            }
+        }
+
+        private string FindHomeSystem(string faction, List<string> factionSystems)
+        {
+            // Look for system which is part of faction name
+            foreach (string system in factionSystems)
+            {
+                string pattern = @"\b" + Regex.Escape(system) + @"\b";
+                if (Regex.IsMatch(faction, pattern)) { return system; }
```

Comment:
Consider skipping the regex?
```cs
{ if faction.Contains(system) }
   { return system; }
}
```

Reply:
I did this deliberately. Contains() will match partials and I require a complete match on the string.